### PR TITLE
CB-18643: Migrate to AWS SDK v2

### PR DESCRIPTION
**Pull request by tiborpopovics**
In this commit, we've replaced the AWS SDK v1 with the v2.
This version change touches several things in our codebase:
- Import changes
- You must create all clients using the client builder method. Constructors are no longer available.
- Request and response object names
- All client class names are now fully camel-cased and no longer prefixed by "Amazon".
- Exception class names, and their structures and relationships, have also changed.
- Client configurations
- Waiters
- Clients and operation request and response objects are now immutable and cannot be changed after creation.
- The setter method names don't include the "set" or "with"

### CODE REVIEW

**Review by bergerdenes — CHANGES_REQUESTED** [review]
Thanks for finishing this large piece.
I have a general ask apart from the small things mentioned: I saw that in some places you migrated to Junit 5 as well, but in some places it is still Junit 4. If it is possible, could you please migrate those tests too?

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/aws/amazonec2/action/EC2ClientActions.java [hunk 1]
```diff
@@ -100,123 +99,138 @@ public List<String> getInstanceVolumeIds(List<String> instanceIds, boolean rootV
...
+
+                    if (waiterResponse.matched().exception().isPresent()) {
+                        throw waiterResponse.matched().exception().get();
+                    }
+
+                    DescribeInstancesResponse describeInstanceResponse = ec2Client.describeInstances(DescribeInstancesRequest.builder()
+                            .instanceIds(instanceId)
+                            .build());
+                    InstanceState actualInstanceState = describeInstanceResponse.reservations().get(0).instances().get(0).state();
+                    if (InstanceStateName.TERMINATED == actualInstanceState.name()) {
```
[the remark this answers is not in the document]
Comment (tiborpopovics) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/aws/amazonec2/action/EC2ClientActions.java:129]: the `actualInstanceState.name()` is returns a InstanceStateName not a string

**Review by bergerdenes — APPROVED** [review]

File: cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/resource/VolumeBuilderUtil.java [hunk 2]
```diff
@@ -53,37 +55,38 @@ private Long getEphemeralCount(AwsInstanceView awsInstanceView) {
...
     public String getRootDeviceName(AuthenticatedContext ac, CloudStack cloudStack) {
         AmazonEc2Client ec2Client = new AuthenticatedContextView(ac).getAmazonEC2Client();
-        DescribeImagesResult images = ec2Client.describeImages(new DescribeImagesRequest().withImageIds(cloudStack.getImage().getImageName()));
-        if (images.getImages().isEmpty()) {
+        DescribeImagesResponse images = ec2Client.describeImages(DescribeImagesRequest.builder().imageIds(cloudStack.getImage().getImageName()).build());
+        if (images.images().isEmpty()) {
             throw new CloudConnectorException(String.format("AMI is not available: '%s'.", cloudStack.getImage().getImageName()));
         }
-        com.amazonaws.services.ec2.model.Image image = images.getImages().get(0);
+        software.amazon.awssdk.services.ec2.model.Image image = images.images().get(0);
```
Comment (lturcsanyi) [inline cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/resource/VolumeBuilderUtil.java:86]: could be an import instead

**Review by lturcsanyi — COMMENTED** [review]
Nice work, minor comments added. 
Could you please elaborate in the description on what/how you have tested?

**Review by lturcsanyi — APPROVED** [review]

**Review by bergerdenes — APPROVED** [review]

**Comment by keyki** [comment]
retest this please

**Comment by tiborpopovics** [comment]
retest this please

**Comment by tiborpopovics** [comment]
retest this please

**Pull request merged**